processor/src/watcher.js

```diff
@@ -82,7 +79,7 @@ function watch(events, repoDir = '/repositories', checkIsRepoReady) {
   return unwatch
 }
 
-async function processRepo(events, repoDir, gitDir) {
+async function processRepo(repoDir, gitDir) {
```

Comment:
I think from this line to the end of the file doesn't belong to the `watcher` we should move it into its own file.

Reply:
I disagree on this one actually. All the watcher does is call processRepo, the rest is all details to get to that point.